debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java

```diff
@@ -324,45 +337,61 @@ protected Object convertBinary(Column column, Field fieldDefn, Object data, Bina
         try {
             if (data instanceof String) {
                 String str = (String) data;
-                if (EMPTY_BLOB_FUNCTION.equals(str)) {
-                    if (column.isOptional()) {
-                        return null;
-                    }
-                    data = "";
-                }
-                else if (isHexToRawFunctionCall(str)) {
+                if (isHexToRawFunctionCall(str)) {
                     data = RAW.hexString2Bytes(getHexToRawHexString(str));
                 }
             }
-            else if (data instanceof Blob) {
-                if (!lobEnabled) {
-                    if (column.isOptional()) {
-                        return null;
-                    }
-                    else {
-                        data = NumberConversions.BYTE_ZERO;
-                    }
-                }
-                else {
-                    Blob blob = (Blob) data;
-                    data = blob.getBytes(1, Long.valueOf(blob.length()).intValue());
-                }
-            }
             else if (data instanceof RAW) {
                 data = ((RAW) data).getBytes();
             }
-
-            if (data == UNAVAILABLE_VALUE) {
```

Comment:
is this if statement not needed for Binary data type? I don't know, maybe not...

Reply:
It was moved to the `convertBlob` method, so it was retained, just moved.